commit 8dcac91a954739b3db29537e7d70d55948e99a8a
Author:     Yuhong Wen <yuhongw@nvidia.com>
AuthorDate: Wed Apr 10 18:52:04 2024 -0400
Commit:     GitHub <noreply@github.com>
CommitDate: Wed Apr 10 18:52:04 2024 -0400

    [2.4] Added more logging for the job status changing (#2481)
    
    * Added more logging for the job status changing.
    
    * Fixed a logging call error.
    
    * Added more logging for the job status change.
    
    ---------
    
    Co-authored-by: Chester Chen <512707+chesterxgchen@users.noreply.github.com>
---
 nvflare/private/fed/server/job_runner.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/nvflare/private/fed/server/job_runner.py b/nvflare/private/fed/server/job_runner.py
index ae2cb7d5..90797f36 100644
--- a/nvflare/private/fed/server/job_runner.py
+++ b/nvflare/private/fed/server/job_runner.py
@@ -443,6 +443,7 @@ class JobRunner(FLComponent):
                                     },
                                     fl_ctx,
                                 )
+                                self.log_info(fl_ctx, f"Updated the schedule history of Job: {job_id}")
 
                             if failed_clients:
                                 deployable_clients = {k: v for k, v in client_sites.items() if k not in failed_clients}
@@ -455,15 +456,19 @@ class JobRunner(FLComponent):
                                 )
                                 continue
 
+                            self.log_info(fl_ctx, f"Start to the run Job: {job_id}")
                             self._start_run(
                                 job_id=job_id,
                                 job=ready_job,
                                 client_sites=deployable_clients,
                                 fl_ctx=fl_ctx,
                             )
+                            self.log_info(fl_ctx, f"Job: {job_id} has been started.")
                             with self.lock:
                                 self.running_jobs[job_id] = ready_job
+                            self.log_info(fl_ctx, f"About to change the Job: {job_id} status to RUNNING.")
                             job_manager.set_status(ready_job.job_id, RunStatus.RUNNING, fl_ctx)
+                            self.log_info(fl_ctx, f"Job: {job_id} started to run, status changed to RUNNING.")
                         except Exception as e:
                             if job_id:
                                 if job_id in self.running_jobs:
